Unknown error

I am receiving the following message when trying to run a test notification for the plugin:

`Unknown errorResponse, please check your Graylog server logs.`

But when I check the Graylog server logs under `/var/log/graylog-server` I find no error messages.

![image](https://user-images.githubusercontent.com/1551020/69664163-69b1af00-104d-11ea-82a5-959852cec5ab.png)

"..." are taking the place of the actual secret strings.

Any ideas? I realize this is not a lot to go on but it is all I have. :)

I am still using the legacy plugin right now..   I tried to upgrade to the 2.0 version and got the same error.   I am using Graylog docker version 3.1.3-1 (also tried most versions between 3.1.0 and 3.1.3).

When you click Execute Test Notification, you see the error "Error: Unknown error, please check your Graylog server logs."

Nothing in the logs but I did notice a 500 error in the post request.

> Internal Server Error
> Reason
> java.lang.NoSuchMethodError: org.graylog.events.notifications.EventNotificationContext.jobTrigger()Lorg/graylog/scheduler/JobTriggerDto;
> 
> Stacktrace
> org.glassfish.jersey.server.ContainerException: java.lang.NoSuchMethodError: org.graylog.events.notifications.EventNotificationContext.jobTrigger()Lorg/graylog/scheduler/JobTriggerDto;
> 	at org.glassfish.jersey.grizzly2.httpserver.GrizzlyHttpContainer$ResponseWriter.rethrow(GrizzlyHttpContainer.java:324)
> 	at org.glassfish.jersey.grizzly2.httpserver.GrizzlyHttpContainer$ResponseWriter.failure(GrizzlyHttpContainer.java:306)
> 	at org.glassfish.jersey.server.ServerRuntime$Responder.process(ServerRuntime.java:509)
> 	at org.glassfish.jersey.server.ServerRuntime$2.run(ServerRuntime.java:334)
> 	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:271)
> 	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:267)
> 	at org.glassfish.jersey.internal.Errors.process(Errors.java:315)
> 	at org.glassfish.jersey.internal.Errors.process(Errors.java:297)
> 	at org.glassfish.jersey.internal.Errors.process(Errors.java:267)
> 	at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:317)
> 	at org.glassfish.jersey.server.ServerRuntime.process(ServerRuntime.java:305)
> 	at org.glassfish.jersey.server.ApplicationHandler.handle(ApplicationHandler.java:1154)
> 	at org.glassfish.jersey.grizzly2.httpserver.GrizzlyHttpContainer.service(GrizzlyHttpContainer.java:384)
> 	at org.glassfish.grizzly.http.server.HttpHandler$1.run(HttpHandler.java:224)
> 	at com.codahale.metrics.InstrumentedExecutorService$InstrumentedRunnable.run(InstrumentedExecutorService.java:181)
> 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
> 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
> 	at java.lang.Thread.run(Thread.java:748)
> Caused by: java.lang.NoSuchMethodError: org.graylog.events.notifications.EventNotificationContext.jobTrigger()Lorg/graylog/scheduler/JobTriggerDto;
> 	at org.graylog.plugins.teams.event.notifications.TeamsEventNotification.getModel(TeamsEventNotification.java:62)
> 	at org.graylog.plugins.teams.event.notifications.TeamsEventNotification.execute(TeamsEventNotification.java:47)
> 	at org.graylog.events.notifications.NotificationResourceHandler.test(NotificationResourceHandler.java:181)
> 	at org.graylog.events.rest.EventNotificationsResource.test(EventNotificationsResource.java:196)
> 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
> 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
> 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
> 	at java.lang.reflect.Method.invoke(Method.java:498)
> 	at org.glassfish.jersey.server.model.internal.ResourceMethodInvocationHandlerFactory$1.invoke(ResourceMethodInvocationHandlerFactory.java:81)
> 	at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher$1.run(AbstractJavaResourceMethodDispatcher.java:144)
> 	at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.invoke(AbstractJavaResourceMethodDispatcher.java:161)
> 	at org.glassfish.jersey.server.model.internal.JavaResourceMethodDispatcherProvider$ResponseOutInvoker.doDispatch(JavaResourceMethodDispatcherProvider.java:160)
> 	at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.dispatch(AbstractJavaResourceMethodDispatcher.java:99)
> 	at org.glassfish.jersey.server.model.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:389)
> 	at org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:347)
> 	at org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:102)
> 	at org.glassfish.jersey.server.ServerRuntime$2.run(ServerRuntime.java:326)
> 	... 14 more

--- a/pom.xml
+++ b/pom.xml
@@ -7,13 +7,13 @@
     <parent>
         <groupId>org.graylog.plugins</groupId>
         <artifactId>graylog-plugin-web-parent</artifactId>
-        <version>3.1.0</version>
+        <version>3.1.3</version>
         <relativePath>../graylog2-server/graylog-plugin-parent/graylog-plugin-web-parent</relativePath>
     </parent>
 
     <groupId>org.graylog.plugins</groupId>
     <artifactId>graylog-plugin-teams</artifactId>
-    <version>2.0.0</version>
+    <version>2.0.1-SNAPSHOT</version>
     <packaging>jar</packaging>
 
     <name>${project.artifactId}</name>
@@ -45,7 +45,7 @@
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
         <maven.compiler.source>1.8</maven.compiler.source>
         <maven.compiler.target>1.8</maven.compiler.target>
-        <graylog.version>3.1.0</graylog.version>
+        <graylog.version>3.1.3</graylog.version>
 
         <!-- Plugins will not be deployed by default - set to `false` if you actually want to deploy it -->
         <maven.source.skip>true</maven.source.skip>

--- a/src/main/java/org/graylog/plugins/teams/event/notifications/TeamsEventNotification.java
+++ b/src/main/java/org/graylog/plugins/teams/event/notifications/TeamsEventNotification.java
@@ -14,6 +14,7 @@ import org.graylog.events.notifications.PermanentEventNotificationException;
 import org.graylog.events.processor.EventDefinitionDto;
 import org.graylog.plugins.teams.client.TeamsClient;
 import org.graylog.plugins.teams.client.TeamsClientException;
+import org.graylog.scheduler.JobTriggerDto;
 import org.graylog2.jackson.TypeReferences;
 import org.graylog2.plugin.MessageSummary;
 
@@ -54,13 +55,14 @@ public class TeamsEventNotification implements EventNotification {
 
   private Map<String, Object> getModel(final EventNotificationContext ctx, final ImmutableList<MessageSummary> backlog) {
     final Optional<EventDefinitionDto> definitionDto = ctx.eventDefinition();
+    final Optional<JobTriggerDto> jobTriggerDto = ctx.jobTrigger();
     final EventNotificationModelData modelData = EventNotificationModelData.builder()
         .eventDefinitionId(definitionDto.map(EventDefinitionDto::id).orElse(UNKNOWN))
         .eventDefinitionType(definitionDto.map(d -> d.config().type()).orElse(UNKNOWN))
         .eventDefinitionTitle(definitionDto.map(EventDefinitionDto::title).orElse(UNKNOWN))
         .eventDefinitionDescription(definitionDto.map(EventDefinitionDto::description).orElse(UNKNOWN))
-        .jobDefinitionId(ctx.jobTrigger().jobDefinitionId())
-        .jobTriggerId(ctx.jobTrigger().id())
+        .jobDefinitionId(jobTriggerDto.map(JobTriggerDto::jobDefinitionId).orElse(UNKNOWN))
+        .jobTriggerId(jobTriggerDto.map(JobTriggerDto::id).orElse(UNKNOWN))
         .event(ctx.event())
         .backlog(backlog)
         .build();

